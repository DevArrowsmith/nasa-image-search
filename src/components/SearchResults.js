import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';


const SearchResultsContainer = styled.div`
  width: calc(100vw - 40px);
  margin: 20px;
  display: flex; 
  flex-flow: row wrap;
  justify-content: center;
  align-items: start;
`;

const ResultsFrame = styled.div`
display: flex; 
flex-flow: row wrap;
justify-content: center;
align-items: start;
`;

const ScaledImage = styled.img`
  width: 270px;
  height: 300px;
  margin: 10px;
  border: none;
  border-radius: 10px;
  overflow: hidden;
  object-fit: cover;
`

const renderImages = (searchResults) => {
    return (
      <ResultsFrame data-testid="images-rendered">
        {searchResults.images.map((result) => (
          <ScaledImage
            key={result.id}
            src={result.image}
            alt={result.description}
          />
        ))}
       </ResultsFrame> 
    )
};


const SearchResults = ({ searchResults }) => {

   return (
      <SearchResultsContainer>
        {renderImages(searchResults)}
      </SearchResultsContainer>
  );
};

SearchResults.propTypes = {
  searchResults: PropTypes.object
}

export default SearchResults;
