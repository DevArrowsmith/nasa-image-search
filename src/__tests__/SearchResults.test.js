import React from 'react';
import { render } from '@testing-library/react';
import SearchResults from '../components/SearchResults';

const mockSearchData = {
  images: [
    {
      id: "id1",
      image: "https://mockaddress.mock/1",
      description: "Mock1"
    },
    {
      id: "id2",
      image: "https://mockaddress.mock/2",
      description: "Mock2"
    },
    {
      id: "id3",
      image: "https://mockaddress.mock/3",
      description: "Mock3"
    },
  ]
};

test('component SearchResults renders', () => {
  const { asFragment } = render(
    <SearchResults searchResults={mockSearchData} />
  );
  const component = asFragment();
  expect(component).toMatchSnapshot();
});

