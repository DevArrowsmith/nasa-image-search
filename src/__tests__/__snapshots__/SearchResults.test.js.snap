// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component SearchResults renders 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm iyNTKJ"
  >
    <img
      alt="Mock1"
      class="sc-AxirZ cxxIYp"
      src="https://mockaddress.mock/1"
    />
    <img
      alt="Mock2"
      class="sc-AxirZ cxxIYp"
      src="https://mockaddress.mock/2"
    />
    <img
      alt="Mock3"
      class="sc-AxirZ cxxIYp"
      src="https://mockaddress.mock/3"
    />
  </div>
</DocumentFragment>
`;
